### ISSUE TITLE
Support installing purchased Apps from the Microsoft Store

### ISSUE THREAD

**Issue by denelon** [microsoft/winget-cli#908]
# Description of the new feature/enhancement

I would like to be able to install Apps that I have already paid for from the Microsoft Store using the Windows Package Manager.

# Proposed technical implementation details (optional)

In order to install paid Apps from the Microsoft Store, the customer needs to be "logged in" with their Microsoft Account (MSA). 

Note: There are likely implications related to the "import" and "export" commands.

**Comment by pb5050** [comment]
how about non purchaased? is it me are none of my app store installed programs listed here?>

**Comment by jedieaston** [comment]
The current Microsoft Store source only shows a specific list of applications that have been added, presumably at some point the source will have all of them.

**Comment by vandreytrindade** [comment]
This is a must have feature. Install/Uninstall Microsoft Store apps (paid or not).

**Comment by remidebette** [comment]
Hi, 

I want to make sure that all my apps are controled by winget now that it is becoming stable.

I cannot install HEVC with winget currently, even though I paid for it in the Store, I guess that it is linked with the current issue?

```powershell
❯ winget install 9N4WGH0Z6VHQ --accept-package-agreements --accept-source-agreements
Found HEVC Video Extensions from Device Manufacturer [9N4WGH0Z6VHQ] Version Unknown
This package is provided through Microsoft Store. winget may need to acquire the package from Microsoft Store on behalf of the current user.
Version: Unknown
Publisher: Microsoft Corporation
Publisher Support Url: https://support.microsoft.com/products/windows?os=windows-10
Description: Play High Efficiency Video Coding (HEVC) videos in any video app on your Windows 10 device. These extensions are designed to take advantage of hardware capabilities on some newer devices— including those with an Intel 7th Generation Core processor and newer GPU to support 4K and Ultra HD content. For devices that don’t have hardware support for HEVC videos, software support is provided, but the playback experience might vary based on the video resolution and PC performance. These extensions also let you encode HEVC content on devices that don’t have a hardware-based video encoder.
License: https://go.microsoft.com/fwlink/?LinkId=822631
Privacy Url: https://privacy.microsoft.com/en-us/privacystatement
Copyright: (c) 2017 Microsoft
Agreements:
Category: Utilities & tools
Pricing: Free
Free Trial: No
Terms of Transaction: https://aka.ms/microsoft-store-terms-of-transaction
Seizure Warning: https://aka.ms/microsoft-store-seizure-warning
Store License Terms: https://aka.ms/microsoft-store-license


Verifying/Requesting package acquisition...
Verifying/Requesting package acquisition failed: server error
```

**Comment by denelon** [comment]
@remidebette,

That's correct. This issue is tracking the request to support installing paid apps where the user already has an entitlement. It will require work on both the WinGet team and the Microsoft Store team.

**Comment by Kein** [comment]
I just realized that Xbox app has winget as a requirement to "install games".
Does that mean WG already has support for subscribed content tied to account?

**Comment by denelon** [comment]
They've integrated with WinGet, but I don't know if they already handle the concept of "provisioned" packages or not.

**Comment by Kein** [comment]
I've just realized this issue isnt part of any Project or Milestone.

**Comment by denelon** [comment]
Quick update. We believe WinGet will be able to install "entitled" apps that one has previously purchased once the "msstore" source has been updated to support packages with entitlements. We're working with the Microsoft Store team on a timeline. We will update this issue once we have an idea when they are likely to prioritize this feature request.

**Comment by Kein** [comment]
What does "entitled" mean in this context and why there is such accent on it?

**Comment by denelon** [comment]
I'm using the term "entitled" to align with the APIs WinGet is calling. If an application has been purchased, the account/user/device would have an "entitlement" for the package/application in question. There are some limitations for some applications that have been "purchased". In some cases, these could be limited to the number of devices the package/application could be installed on.

We're not expecting WinGet to be involved in a "purchase" flow, but assuming an application/package has been purchased, then WinGet should be able to install applications based on the context of the user/device when WinGet is used to request the installation of a package/application via the Microsoft Store.

**Comment by Kein** [comment]
How is the authentication part will be done? What will it use to verify user?
Let's assume case Windows Store is never installed.

**Comment by denelon** [comment]
We call a generic entitlement (only good for "Free" and rated "e") for the device if no user is logged into the Microsoft Store. We're anticipating multiple future scenarios.

**Comment by Kein** [comment]
> if no user is logged into the Microsoft Store.
That is what I'm asking about. Logged in how? Exactly via MS Store app? Via OS system-wide login? Or winget will have its own auth mechanism?

**Comment by albintho** [comment]
Is this request completed? it would be better if options are available on winget-cli itself to verify the credentials rather than asking users to log through mstore and install apps through winget.

**Comment by Itsamejonah** [comment]
```python
print"patrick"
```

**Comment by Obendorf96** [comment]
is there any Update on this issue ?

**Comment by aubymori** [comment]
bump

**Comment by peperizal1133-afk** [comment]
Jose Rizal has sent you an email via Gmail confidential mode:

[image: Gmail logo]Re: [microsoft/winget-cli] Support installing purchased Apps from the Microsoft Store (#908) <https://confidential-mail.google.com/msg/AJ05Yhf24kYX50cjNv6f6IZpKcqQwmiXRXsPJKbRojrvKeZvku-U5N4IsMCaJE6Ip_B-SRZ6i9HFW1HPvflrDrPkBiR1w5GsTQod5Kh56AYu2kqTaAWJNkoVK8v0X8c9by77ytncH2IZ8kQRQXkJzScFmuH_R0jOt8NCUMhesXhBY3BMs8aYPV06RPfO8y1rcAzBE4MFMpvzu6V2JHc=>

This message was sent on Jan 30, 2026 at 12:15:28 AM PST
You can open it by clicking the link below. This link will only work for ***@***.***

View the email <https://confidential-mail.google.com/msg/AJ05Yhf24kYX50cjNv6f6IZpKcqQwmiXRXsPJKbRojrvKeZvku-U5N4IsMCaJE6Ip_B-SRZ6i9HFW1HPvflrDrPkBiR1w5GsTQod5Kh56AYu2kqTaAWJNkoVK8v0X8c9by77ytncH2IZ8kQRQXkJzScFmuH_R0jOt8NCUMhesXhBY3BMs8aYPV06RPfO8y1rcAzBE4MFMpvzu6V2JHc=>

Gmail confidential mode gives you more control over the messages you send. The sender may have chosen to set an expiration time, disable printing or forwarding, or track access to this message. Learn more <https://support.google.com/mail/answer/7674059>
Gmail: Email by Google
Use is subject to the Google Privacy Policy <https://myaccount.google.com/privacypolicy?hl=en>
Google LLC, 1600 Amphitheatre Parkway, Mountain View, CA 94043, USA
You have received this message because someone sent you an email via Gmail confidential mode.
[image: Google logo]
